pywren/future.py

```diff
@@ -172,36 +187,34 @@ def result(self, timeout=None, check_only=False, throw_except=True, storage_hand
 
         self._call_invoker_result = call_invoker_result
 
-
-
         if call_success:
 
             self._return_val = call_invoker_result['result']
             self._set_state(JobState.success)
             return self._return_val
+        else:
+            self._set_state(JobState.error)
+            if throw_except:
 
-        elif throw_except:
-
-            self._exception = call_invoker_result['result']
-            self._traceback = (call_invoker_result['exc_type'],
-                               call_invoker_result['exc_value'],
-                               call_invoker_result['exc_traceback'])
-
-            self._state = JobState.error
```

Comment:
We seem to have dropped this line `self._state = JobState.error` -- is this intentional ?

Reply:
we switched to using the proper `self._set-state(JobState.error)` in line 195